src/config.rs

```diff
@@ -81,6 +82,19 @@ impl Default for Config {
     }
 }
 
+impl std::fmt::Display for Config {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+        write!(fmt, "(user={} directory={} file={} users={} size={} generations={})", 
+                self.user.clone().unwrap_or("n/a".to_string()), 
+                self.directory.clone().unwrap_or(env::current_dir().unwrap()).display(), 
```

Comment:
Please get rid of the `unwrap`. Formatting the possible error message using `or_else` should be fine.

Reply:
Do you mean something like
```rust
self.directory.clone().unwrap_or(std::env::current_dir().or_else(|e| Err(e))?).display(),
```
But like this, it does not work, since std::io::Error can't be converted to std::fmt::Error. Any hint on what I should do?

Most easy/straightforward could be to change the line to:
```rust
self.directory.clone().unwrap_or(env::current_dir().unwrap_or(PathBuf::from("."))).display(),
```

Reply:
I have tinkered around a bit before coming to the conclusion that displaying an error is not that important. The default value isn't of much value in production anyway. How about simply doing this?
``` rs
self.directory.clone().unwrap_or_else(||Path::new(".").into()).display()
```
but etither way is fine.

Reply:
Thanks for the guidance.  I think this is sufficient logging, the cwd can be easily inferred from other log messages, if needed. So, I changed it in 484976de1eb691dca4ccfb352c95af129c975ee4, only difference is, that I sticked to `PathBuf` to be constistent with the `Option<PathBuf>`. Hope, this is fine for you.